web/src/infra/i18n/hocs/withTranslate.tsx

```diff
@@ -0,0 +1,35 @@
+import { getDisplayName } from 'recompose';
+import { nextI18Next } from '../config';
+import React from 'react';
+import { useCurrentGame } from 'infra/game/GameProvider';
+
+export const withTranslate = (propertyName: string = 'translate', options: { withRef?: boolean } = {}) => {
+  return function Extend(WrappedComponent) {
+    function I18nextWithTranslation({ forwardedRef, ...rest }) {
+      const { game } = useCurrentGame();
+      const ns = game?.code;
+      const [t] = nextI18Next.useTranslation(ns, rest);
```

Comment:
This seems great, by using I18Next `useTranslation` it seems like you will be avoiding loading other langauges files, right? So the user seeing the page in portuguese wouldn't have to download the description of the game in french, for instance :)

Reply:
Yeah! `useTranslation(ns, ...)` loads a namespace with name = `game.code`, but only for the current language.